packages/common/src/transports/wiresaw.ts

```diff
@@ -0,0 +1,188 @@
+          }
+
+          if (req.method === "eth_estimateUserOperationGas") {
+            return await estimateUserOperationGas({
+              request: wiresawRequest,
+              params: req.params as never,
+            });
+          }
+
+          // Fallback to regular RPC for methods that don't require wiresaw
+          if (
+            req.method === "eth_blockNumber" ||
+            req.method === "eth_getBlockByNumber" ||
+            req.method === "eth_maxPriorityFeePerGas"
+          ) {
+            if (fallbackDefaultTransport) {
+              const { request: fallbackRequest } = fallbackDefaultTransport(opts);
+              return await fallbackRequest(req);
+            }
+            return await wiresawRequest(req);
+          }
+
+          return await wiresawRequest(req);
+        } catch (e) {
+          console.warn("[wiresaw] request failed", e);
+          const bundlerMethods = [
+            "eth_estimateUserOperationGas",
+            "eth_sendUserOperation",
+            "eth_getUserOperationReceipt",
+          ];
+          if (bundlerMethods.includes(req.method)) {
+            if (fallbackBundlerTransport) {
+              const { request: fallbackRequest } = fallbackBundlerTransport(opts);
+              console.warn("[wiresaw] falling back to bundler rpc", req);
+              return fallbackRequest(req);
+            }
+          }
+          if (fallbackDefaultTransport) {
+            const { request: fallbackRequest } = fallbackDefaultTransport(opts);
+            console.warn("[wiresaw] falling back to eth rpc", req);
+            return fallbackRequest(req);
+          }
+          throw e;
+        }
+
+        async function getTransactionReceipt(hash: Hex): Promise<RpcTransactionReceipt | undefined> {
+          // Return cached receipt if available
+          if (transactionReceipts[hash]) return transactionReceipts[hash];
+
+          // Fetch pending receipt
+          const pendingReceipt = (await wiresawRequest({
+            ...req,
+            method: "wiresaw_getTransactionReceipt",
+            params: [hash],
+          })) as RpcTransactionReceipt | undefined;
+          if (pendingReceipt) {
+            transactionReceipts[hash] = pendingReceipt;
+            return pendingReceipt;
+          }
+
+          if (fallbackDefaultTransport) {
+            const { request: fallbackRequest } = fallbackDefaultTransport(opts);
+            const receipt = (await fallbackRequest({
+              ...req,
+              method: "eth_getTransactionReceipt",
+              params: [hash],
+            })) as RpcTransactionReceipt | undefined;
+            if (receipt) {
+              transactionReceipts[hash] = receipt;
+              return receipt;
+            }
+          }
+        }
+      },
+    };
+  }) as wiresawTransport;
+}
+
+function getWiresawBaseTransport(chain: Chain): Transport | undefined {
```

Comment:
not blocking, but we could probably generalize this to something like `chainTransport` given a chain's RPC URLs object like `chainTransport(chain.rpcUrls.wiresaw)`, since there are a bunch of areas we use this pattern where we want websocket + http if one or both exist